packages/webapp/src/journey-validation/allocation/applicant-details.js

```diff
@@ -52,7 +65,24 @@ const WRITTEN_AUTHORISATION_UPLOAD = routeDefinition(
 
 const WRITTEN_AUTHORISATION_CHECKED = routeDefinition(
   constants.routes.DEVELOPER_CHECK_WRITTEN_AUTHORISATION_FILE,
-  [constants.redisKeys.DEVELOPER_WRITTEN_AUTHORISATION_CHECKED]
+  [constants.redisKeys.DEVELOPER_WRITTEN_AUTHORISATION_CHECKED],
+  (session) => {
+    const checkWrittenAuthorisation = session.get(constants.redisKeys.DEVELOPER_WRITTEN_AUTHORISATION_CHECKED)
+    if (checkWrittenAuthorisation === 'no') {
+      return constants.routes.DEVELOPER_UPLOAD_WRITTEN_AUTHORISATION
+    } else if (checkWrittenAuthorisation === 'yes') {
+      const isLandowner = session.get(constants.redisKeys.DEVELOPER_LANDOWNER_OR_LEASEHOLDER) === constants.DEVELOPER_IS_LANDOWNER_OR_LEASEHOLDER.YES
+      if (!isLandowner) {
+        return constants.routes.DEVELOPER_UPLOAD_CONSENT_TO_ALLOCATE_GAINS
+      }
+      return session.get(constants.redisKeys.PRIMARY_ROUTE) || constants.routes.DEVELOPER_TASKLIST
```

Comment:
Example of returning the primary route - controlled at this level rather than trying to calculate if a journey is complete because this would be the last step of a journey.